app/code/Magento/Sales/Model/Order/Payment/Info.php

```diff
@@ -98,6 +104,23 @@ public function getData($key = '', $index = null)
         return parent::getData($key, $index);
     }
 
+    /**
+     * Get the current order to manipulate order details
+     *
+     * @return mixed
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    protected function getOrder()
```

Comment:
Hi @sarfarazbheda. Thank you for your proposal. Unfortunately, we cannot rely on the coreRegistry here because of a couple of reasons.

1. This is a very situational solution and will work only in case if the corresponding value in the Registry is set. There's a high chance that the Registry will not have this value in the further versions (read below).
2. The Registry concept has been deprecated from v2.2.

Ideally, we should pass the `$storeId` to the `getMethodInstance()`